jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java

```diff
@@ -513,22 +519,30 @@ protected Client getWSClient(WorkItem workItem, String interfaceRef) {
 	                                                                                  interfaceRef),
 	                                                                        getInternalClassLoader(),
 	                                                                        null);
-	                        setClientTimeout(workItem, client);
-	                        clients.put(interfaceRef,
-	                                    client);
-	                        return client;
-	                    } catch (Exception e) {
-	                        logger.error("Error when creating WS Client",
-	                                     e);
-	                        continue;
-	                    }
-	                }
-	            }
-	        }
+                            setClientTimeout(workItem, client);
+                            Collection<WorkItemHeaderInfo> headers = WorkItemHeaderUtils.getHeaderInfo(workItem);
+                            if (!headers.isEmpty()) {
+                                client.getRequestContext().put(Header.HEADER_LIST, headers.stream().map(
+                                        this::buildHeader).collect(Collectors.toList()));
+                            }
+                            clients.put(interfaceRef, client);
+                            return client;
+                        } catch (Exception e) {
+                            logger.error("Error when creating WS Client", e);
+                        }
+                    }
+                }
+            }
         }
         return null;
     }
 
+	private Header buildHeader(WorkItemHeaderInfo header) {
+		String namespace = (String) header.getParam("NS");
+		QName name = namespace == null ? new QName(header.getName()) : new QName(namespace, header.getName());
+		return new Header(name, header.getContent());
```

Comment:
It's needed to use the constructor with the DataBinding (a ClassCastException is thrown otherwise), and handle the JAXBException:

```suggestion
		return new Header(name, header.getContent(), new JAXBDataBinding(String.class));
```

Reply:
I take the change to put that class as configurable through a header param